File: clients/web/src/app/shared/utils/record-lifecycle.util.ts
/** Soft-delete lifecycle helpers — shared by entity page and record header. */

export function isRecordDeleted(record: Record<string, unknown>): boolean {
  const deletedAt = record['deleted_at'];
  return deletedAt !== undefined && deletedAt !== null && String(deletedAt).length > 0;
}

export function canDeleteRecord(
  recordId: string | null,
  record: Record<string, unknown>,
  creatingNew: boolean,
): boolean {
  return Boolean(recordId && !creatingNew && !isRecordDeleted(record));
}

export function canRestoreRecord(
  recordId: string | null,
  record: Record<string, unknown>,
): boolean {
  return Boolean(recordId && isRecordDeleted(record));
}

File: clients/mobile/lib/utils/record_lifecycle_util.dart
/// Soft-delete lifecycle helpers — web `record-lifecycle.util.ts` parity.

bool isRecordDeleted(Map<String, dynamic> record) {
  final deletedAt = record['deleted_at'];
  if (deletedAt == null) return false;
  return '$deletedAt'.trim().isNotEmpty;
}

bool isRecordLoaded(String? recordId, Map<String, dynamic> record) {
  if (recordId == null || recordId.isEmpty) return false;
  return '${record['id'] ?? ''}' == recordId;
}

bool canDeleteRecord(String? recordId, Map<String, dynamic> record, bool creatingNew) {
  return isRecordLoaded(recordId, record) && !creatingNew && !isRecordDeleted(record);
}

bool canRestoreRecord(String? recordId, Map<String, dynamic> record) {
  return isRecordLoaded(recordId, record) && isRecordDeleted(record);
}
